File: DWSIM.ProFeatures/FormCosting.vb
Imports DWSIM.Interfaces

Public Class FormCosting

    Inherits WeifenLuo.WinFormsUI.Docking.DockContent

    Public CurrentFlowsheet As IFlowsheet

    Private Sub FormCosting_Load(sender As Object, e As EventArgs) Handles MyBase.Load

        ExtensionMethods.ChangeDefaultFont(Me)

    End Sub

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click

        Functions.CreateTransitionObject(CurrentFlowsheet, "", "Costing", "", "", Nothing)

        Functions.ProcessTransition(CurrentFlowsheet)

    End Sub

End Class

File: DWSIM.ProFeatures/FormCosting.cs
using DWSIM.Interfaces;
using System;

namespace DWSIM.ProFeatures
{


    public partial class FormCosting : WeifenLuo.WinFormsUI.Docking.DockContent
    {

        public IFlowsheet CurrentFlowsheet;

        public IAnalyticsProvider AnalyticsProvider;

        public FormCosting()
        {
            InitializeComponent();
            lblFeature = _lblFeature;
            _lblFeature.Name = "lblFeature";
        }

        private void Button2_Click(object sender, EventArgs e)
        {

            Functions.CreateTransitionObject(CurrentFlowsheet, "", "Costing", "", "", default);

            Functions.DisplayTransitionForm(AnalyticsProvider, CurrentFlowsheet, "Costing");

        }

        private void FormCosting_Load_1(object sender, EventArgs e)
        {
            ExtensionMethods.FormExtensions.ChangeDefaultFont(this);
        }
    }
}